compiler/tests/should-pass/code-generation/functions/Arrays.via

```diff
@@ -0,0 +1,37 @@
+host alice: {A}
+
+fun main() {
+    val a = Array[int]{A}(1);
+    output a[0] to alice;  /* expects 0 */
+
+    /* Passing array obj ref as in arg */
+    /* Modifying in arg array */
+    f0(&a);
+    output a[0] to alice;  /* expects 1 */
+
+    /* Passing array declaration as out arg */
+    /* Passing out param array as arg to new fn call */
+    f1(val b);
+    output b[0] to alice;  /* expects 0 */
+
+    /* Initializing out param array, then modifying array and returning */
+    f2(val d);
+    output d[0] to alice;  /* expects 2 */
+}
+
+fun f0(a: Array[int]) {
+    a[0] = 1;
+}
+
+fun f1(b: out Array[int]{A}) {
+    g(out b);
```

Comment:
Try accessing `b` right after the call to `g` (e.g., b[0] *= 2).. This will cause an error in generated code since `b_boxed_ is defined, but `b` isn't.

Reply:
Hacky soln: Initialize local vars after fn calls which involve OutParamArgs. Kind of messy. This ok?

```
fun f1(b: out Array[int]{A}) {
    g(out b);
    b[0] = 5;
}
```
compiles to    
```
private fun f1_1(b_1_box: Out<Array<Int>>): Unit {
      val b_1: Array<Int>
      g_1(b_1_box)
      b_1 = b_1_box.get()
      // b_1[0] = 5
      b_1[0] = 5
}
```

Reply:
Calling a function with `OutParamArgs` should have the same effect as an `OutParameterInitializationNode`. This change accomplishes that, right? So I don't think it's that hacky.